cmd/axiom-syslog-proxy/main.go

```diff
@@ -1,52 +1,79 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"log"
 	"os"
+	"os/signal"
+	"syscall"
 
 	"github.com/axiomhq/axiom-go/axiom"
 	"github.com/axiomhq/pkg/version"
 
 	"github.com/axiomhq/axiom-syslog-proxy/server"
 )
 
-var (
-	deploymentURL = os.Getenv("AXIOM_URL")
-	ingestToken   = os.Getenv("AXIOM_TOKEN")
-	ingestDataset = os.Getenv("AXIOM_DATASET")
+const (
+	exitOK int = iota
+	exitConfig
+	exitInternal
+)
 
+var (
 	addrTCP = flag.String("addr-tcp", ":601", "Listen address <ip>:<port>")
 	addrUDP = flag.String("addr-udp", ":514", "Listen address <ip>:<port>")
 )
 
 func main() {
+	os.Exit(Main())
+}
+
+func Main() int {
```

Comment:
Can we think of a better name? `run` maybe?

Reply:
No. The other PRs went through already and I have seen this practice elsewhere.

Reply:
I hate this but ok